### PR TITLE
docs(client): update Linux and Windows Client docs

### LINKED ISSUE #4925: Linux GUI client docs
Update /kb/user-guides/linux-client with GUI client instructions, requirements, etc

### LINKED ISSUE #4963: Usage and uninstall docs for all Clients
Uninstall / usage docs for Clients, in the existing `/kb/user-guides`. For the GUI clients, we should assume a non-technical user might be reading them. For the headless/CLI clients, we can assume the user has basic proficiency of the CLI.

Should cover:

- Prereqs / requirements
- How to install
- How to operate (wait to take screenshots until #3514 is complete, since menus will be changing in that)
  - Signing in
  - "You should see Resources you can access in the menu" -- one user hit this
  - Signing out
  - Troubleshooting / known issues
  - Viewing and clearing logs

```[tasklist]
### Tasks
- [ ] https://github.com/firezone/firezone/issues/4995
- [ ] https://github.com/firezone/firezone/issues/4925
- [ ] https://github.com/firezone/firezone/issues/4997
- [ ] https://github.com/firezone/firezone/issues/4998
- [ ] https://github.com/firezone/firezone/issues/4999
- [ ] https://github.com/firezone/firezone/issues/5000
```

### LINKED ISSUE #4995: Usage docs for Windows
See parent issue for details

### LINKED ISSUE #4997: Update usage docs for Linux headless
Currently have a decent Linux headless guide, just needs to be updated with recent changes/refactors.

### LINKED ISSUE #5047: bug(gui-client/Linux): error dialog is still showing "adduser" instead of "usermod"
Should be `sudo usermod -a -G firezone-client $USER`

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #4995
Closes #4925
Closes #4997
Closes #5047 
Supersedes #4965 and #5004.

NOT changing:

- Page description for other Clients. That is still "Firezone Documentation"

Need these Clients:

- Windows GUI
- Linux headless 
- Linux GUI

to have these things documented: (with exact terms)

- Prerequisites
- Installation
- Usage
  - Signing in
  - Accessing a Resource
  - Signing out
  - Quitting
- Upgrading
- Diagnostic logs
- Uninstalling
- Troubleshooting
  - DNS not reverted after exit
  - DNS Resource not accessible
- Known issues

```[tasklist]
### Before merging
- [x] Test Windows GUI instructions
- [x] Add troubleshooting for #5027
- [x] Fill in troubleshooting sections
- [x] Test Linux GUI instructions
- [x] Linux headless - Make sure SIGTERM or Ctrl+C or whatever reverts resolv.conf
- [x] Test Linux Headless instructions
- [x] Page descriptions should be "How to install and use the Firezone $OS $UI client."
- [x] ~~Linux headless - Confirm behaviors and default values of all env vars~~ (skipping - The ones that are used are exercised)
- [x] Grep for TODOs
- [x] Change "un-install" to "uninstall"
- [x] Capitalize "Client" where needed
- [x] Change "IPC service" to "Tunnel service" or something
- [x] Change "SplitDNS" to "Split DNS"
- [ ] Wait for next Client release to be cut
```

### CODE REVIEW

**Comment by ReactorScram** [comment]
@jamilbk Does the overall structure look right? I set up the same sections for all 3 pages, mostly filled them in, and changed some "you can download from here" type sentences to imperative, "Download from here".

All that's left is filling in troubleshooting and then testing everything step-by-step, which will take a while and hopefully result in only minor changes.

**Comment by jamilbk** [comment]
@ReactorScram Looks great!

**Comment by ReactorScram** [comment]
I tested the Windows instructions using my Windows Server 2022 VM and the tip of the Windows IPC PR and it works well.

**Comment by ReactorScram** [comment]
@jamilbk I tested everything else and made a small code fix too, looks like `FIREZONE_NAME` got lost somewhere in the headless Client, so I added it back in. It's ready for review

File: website/src/app/kb/user-guides/linux-client/readme.mdx
Comment (jamilbk) [inline website/src/app/kb/user-guides/linux-client/readme.mdx:37]: I know it usually makes sense to add language specifiers to these code blocks, but doing so makes formatting a little ugly since it adds `<code>` tags that don't have the right CSS configured for them. The fix requires some digging into React/react-markdown/hljs and Flowbite's `format` typography plugin.

Opened this to track fix: https://github.com/firezone/firezone/issues/5057

Screenshot 2024-05-20 at 7 12 07 PM

File: website/src/app/kb/user-guides/linux-gui-client/readme.mdx
Comment (jamilbk) [inline website/src/app/kb/user-guides/linux-gui-client/readme.mdx:32]: styling

File: website/src/app/kb/user-guides/linux-client/readme.mdx
Comment (jamilbk) [inline website/src/app/kb/user-guides/linux-client/readme.mdx:78]: I still worry about this UX as I think it's going to surprise nearly everyone that runs the headless client. Maybe the next thing to work on.

**Review by jamilbk — APPROVED** [review]

**Comment by jamilbk** [comment]
Let's wait to merge until we cut the next release (tomorrow?)

File: website/src/app/kb/user-guides/linux-client/readme.mdx
Comment (ReactorScram) [reply website/src/app/kb/user-guides/linux-client/readme.mdx:78]: I'd be okay with bailing out if the user doesn't specify any DNS control. Alternately we could make systemd-resolved the default, but then that default becomes part of the CLI interface forever, not sure if I like that.

File: website/src/app/kb/user-guides/linux-client/readme.mdx
Comment (ReactorScram) [reply website/src/app/kb/user-guides/linux-client/readme.mdx:37]: ~~I'll merge this after that code block fix merges then~~ I'll wait to merge this since you said website changes need extra coordination

**Comment by jamilbk** [comment]
I'll resolve conflicts and merge this now.

**Pull request merged**
